src/GitHubExtension/Widgets/GitHubUserWidget.cs

```diff
@@ -0,0 +1,348 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using System.Text.Json.Nodes;
+using GitHubExtension.DataManager;
+using GitHubExtension.DeveloperId;
+using GitHubExtension.Helpers;
+using GitHubExtension.Widgets.Enums;
+using Microsoft.Windows.DevHome.SDK;
+using Microsoft.Windows.Widgets.Providers;
+using Octokit;
+
+namespace GitHubExtension.Widgets;
+internal abstract class GitHubUserWidget : GitHubWidget
+{
+    protected static readonly new string Name = nameof(GitHubUserWidget);
+
+    protected string DeveloperLoginId { get; set; } = string.Empty;
+
+    protected SearchCategory ShowCategory { get; set; } = SearchCategory.Unknown;
+
+    private string userName = string.Empty;
+
+    protected string UserName
+    {
+        get
+        {
+            if (string.IsNullOrEmpty(userName))
+            {
+                GetUserName();
+            }
+
+            return userName;
+        }
+        set => userName = value;
+    }
+
+    public GitHubUserWidget()
+        : base()
+    {
+        GitHubSearchManager.OnResultsAvailable += SearchManagerResultsAvailableHandler;
+        ShowCategory = SearchCategory.Unknown;
+        UserName = string.Empty;
+    }
+
+    ~GitHubUserWidget()
+    {
+        GitHubSearchManager.OnResultsAvailable -= SearchManagerResultsAvailableHandler;
+    }
+
+    protected abstract string GetTitleIconData();
+
+    private void GetUserName()
+    {
+        var devIds = DeveloperIdProvider.GetInstance().GetLoggedInDeveloperIdsInternal();
+        if ((devIds != null) && devIds.Any())
+        {
+            userName = devIds.First().LoginId;

```

Comment:
Does this get overwritten somewhere, or are we just always using the first one in the list?

Reply:
Fixed.